components/browser/icons/src/test/java/mozilla/components/browser/icons/loader/HttpIconLoaderTest.kt

```diff
@@ -196,4 +197,31 @@ class HttpIconLoaderTest {
 
         assertEquals(IconLoader.Result.NoResult, loader.load(testContext, mock(), resource))
     }
+
+    @Test
+    fun `Loader will return NoResult for IOExceptions happening during toIconLoaderResult`() {
+        val client: Client = mock()
+
+        val failingStream: InputStream = object : InputStream() {
+            override fun read(): Int {
+                throw IOException("Kaboom")
+            }
+        }
```

Comment:
Another option here is to use [Mockito](https://site.mockito.org/) to create a mocked `InputStream` that throws:

```Kotlin
val failingStream: InputStream = mock()
doThrow(IOException("Kaboom")).when(failingStream).read()
```

But creating your own stream is nice too!